cartesian_trajectory_controller/include/cartesian_trajectory_controller/control_policies.h

```diff
@@ -187,7 +190,8 @@ namespace cartesian_ros_control
 
 
       private:
-        std::unique_ptr<KDL::ChainIkSolverPos_LMA> ik_solver_;
+        std::unique_ptr<pluginlib::ClassLoader<IKSolver> > solver_loader_;
+        boost::shared_ptr<IKSolver> ik_solver_;
```

Comment:
The mixing of `std::unique_ptr` and `boost::shared_ptr` next to each other reads a bit weird. One possible solution would be to store the `ik_solver_` inside an `std` pointer and use [`createUnmanagedInstance()`](http://docs.ros.org/en/noetic/api/pluginlib/html/classpluginlib_1_1ClassLoader.html#ad0774d62234179c16f637184da987da8) to create the instance.

Reply:
Awesome! Thanks @fmauch for `createUnmanagedInstance()`! I overlooked that possibility and didn't like the mixed appearance either. See, a2c0204